src/backend/cuda/kernel/meanshift.hpp

```diff
@@ -12,200 +12,152 @@
 #include <Param.hpp>
 #include <debug_cuda.hpp>
 #include <math.hpp>
-#include "shared.hpp"
+
+#include <type_traits>
 
 namespace cuda
 {
-
 namespace kernel
 {
+static const int THREADS_X = 32;
```

Comment:
did you see a noticeable speedup with these parameters? It might be better to make this change in another PR.